src/flib/pio_types.F90

```diff
@@ -171,6 +171,11 @@ module pio_types
 !!  - PIO_int :  4-byte integers
 !!  - PIO_char : character
 !<
+   integer, public, parameter :: PIO_MAX_DIMS_UB = 1024
+   integer, public, parameter :: PIO_MAX_ATTRS_UB = 8192
+   integer, public, parameter :: PIO_MAX_VARS_UB = 8192
+   integer, public, parameter :: PIO_MAX_NAME_UB = 256
```

Comment:
Again, I think we should increase PIO_MAX_NAME upper bound to at least 1024 (this is most likely the limit that we will hit for E3SM).